### PR TITLE
skip node reboot test if provider is local.

### LINKED ISSUE #72225: Test to reboot the node should be skipped when using local-up-cluster.sh.
**What happened**:
Run the following test using the cluster set up by local-up-cluster.sh will reboots own node.

[It] each node by ordering clean reboot and ensure they function upon restart
[It] each node by ordering unclean reboot and ensure they function upon restart

**What you expected to happen**:
The mentioned tests should be skipped when using local-up-cluster.sh.

**How to reproduce it (as minimally and precisely as possible)**:
Run the mentioned tests using the cluster set up by local-up-cluster.sh.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release): Ubuntu 18.04.1
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/kind bug

### PR DESCRIPTION
**Pull request by s-ito-ts**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Skip node reboot test using the cluster set up by local-up-cluster.sh (local provider).
Because It will reboot the node that ran e2e test.

**Which issue(s) this PR fixes**:
Fixes #72225 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by oomichi** [comment]
Even if both e2e test runner and k8s-apiserver exist on the same node as "local" provider, the test runner node could not be rebooted and the problem could not happen if k8s-apiserver node is not marked as schedulable.
In most cases of "local" provider, the k8s cluster would be all-in-one on a single node and this problem tends to happen easily. So this change can be useful for most cases.

/lgtm

**Comment by BenTheElder** [comment]
local generally means hack/local-up-cluster.sh, we don't want to reboot the host!

File: test/e2e/lifecycle/reboot.go
Comment (BenTheElder) [inline test/e2e/lifecycle/reboot.go]: I think this already skips "skeleton" 🤔
We have other local clusters that set skeleton instead of local for conformance testing &/or avoiding some of the baggage local has with invoking the scripts in cluster/

Cc @dims

**Review by BenTheElder — APPROVED** [review]
/lgtm
/hold
Would like @dims's thoughts as the effective maintainer of local-up-cluster

**Comment by dims** [comment]
i don't want to special case "local" in the test suite. let's please use the regex for the reboot feature to skip for specific ci jobs.

**Comment by BenTheElder** [comment]
> i don't want to special case "local" in the test suite. let's please use the regex for the reboot feature to skip for specific ci jobs.

Is this marked [Feature:NodeReboot]?
Having this in the default suite would be problematic for other providers as well.

**Comment by smarterclayton** [comment]
I can't think of any case where this test should be executed on a local provider.  @dims why do you want this to be an exclusion rule?

**Comment by BenTheElder** [comment]
> I can't think of any case where this test should be executed on a local provider. @dims why do you want this to be an exclusion rule?

Minor note: "local" is specifically treated as hack/local-up-cluster.sh and has the log dump hack enabled in the e2e suite. In general newly named providers are not recognized by the suite now (and it will fail) so almost everyone has to be provider = "skeleton" now unless they were one of the originals.

**Comment by dims** [comment]
@smarterclayton +1 to what @BenTheElder said.

**Pull request closed without merging**
